test/unit/nic_test.rb

```diff
@@ -58,13 +58,61 @@ def teardown
   end
 
   test "should delegate subnet attributes" do
-    subnet = subnets(:one)
+    subnet = subnets(:two)
     domain = (subnet.domains.any? ? subnet.domains : subnet.domains << Domain.first).first
     interface = Nic::Managed.create! :ip => "2.3.4.127", :mac => "cabbccddeeff", :host => FactoryGirl.create(:host), :subnet => subnet, :name => "a" + FactoryGirl.create(:host).name, :domain => domain
     assert_equal subnet.network, interface.network
     assert_equal subnet.vlanid, interface.vlanid
   end
 
+  test "should reject subnets with mismatched organization in host" do
+    organization1 = FactoryGirl.create(:organization)
+    oragnization2 = FactoryGirl.create(:organization)
+    subnet = subnets(:one)
+    host = FactoryGirl.create(:host)
+
+    subnet.organizations << organization1
+    host.organization = oragnization2
+
+    i = Nic::Base.new :mac => "cabbccddeeff", :host => host
+    i.subnet = subnet
+
+    assert_not i.valid?
+  end
+
+  test "should reject subnets with mismatched location in host" do
```

Comment:
Is this what you're looking for? It's DRY and it's `#build`ing rather than `#create`-ing
```Ruby
  test "should reject subnet with mismatched taxonomy in host" do
    taxonomy_to_test = [:organization, :location]

    taxonomy_to_test.each do |taxonomy|
      tax_object1 = FactoryGirl.build(taxonomy)
      tax_object2 = FactoryGirl.build(taxonomy)
      subnet = subnets(:one)
      host = FactoryGirl.build(:host)

      subnet_list = subnet.send("#{taxonomy.to_s.pluralize}")
      subnet_list << tax_object1
      host.send("#{taxonomy}=",tax_object2)

      i = Nic::Base.new :mac => "cabbccddeeff", :host => host
      i.subnet = subnet

      refute i.valid?
    end
  end
```